pyiceberg/table/snapshots.py

```diff
@@ -116,3 +144,202 @@ class MetadataLogEntry(IcebergBaseModel):
+
+        return properties
+
+
+def truncate_table_summary(summary: Summary, previous_summary: Mapping[str, str]) -> Summary:
+    for prop in {
+        'total-data-files',
+        'total-delete-files',
+        'total-records',
+        'total-files-size',
+        'total-position-deletes',
+        'total-equality-deletes',
+    }:
+        summary[prop] = '0'
+
+    if value := previous_summary.get('total-data-files'):
+        summary['deleted-data-files'] = value
+    if value := previous_summary.get('total-delete-files'):
+        summary['removed-delete-files'] = value
+    if value := previous_summary.get('total-records'):
+        summary['deleted-records'] = value
+    if value := previous_summary.get('total-files-size'):
+        summary['removed-files-size'] = value
+    if value := previous_summary.get('total-position-deletes'):
+        summary['removed-position-deletes'] = value
+    if value := previous_summary.get('total-equality-deletes'):
+        summary['removed-equality-deletes'] = value
+
+    return summary
+
+
+def merge_snapshot_summaries(
+    summary: Summary,
+    previous_summary: Optional[Mapping[str, str]] = None,
+) -> Summary:
+    if summary.operation not in {Operation.APPEND, Operation.OVERWRITE}:
+        raise ValueError(f"Operation not implemented: {summary.operation}")
+
+    if summary.operation == Operation.OVERWRITE and previous_summary is not None:
+        summary = truncate_table_summary(summary, previous_summary)
+
+    if not previous_summary:
+        previous_summary = {
+            'total-data-files': '0',
+            'total-delete-files': '0',
+            'total-records': '0',
+            'total-files-size': '0',
+            'total-position-deletes': '0',
+            'total-equality-deletes': '0',
+        }
+
+    def _update_totals(total_property: str, added_property: str, removed_property: str) -> None:
+        if new_total_str := previous_summary.get(total_property):
+            new_total = int(new_total_str)
+            if new_total >= 0 and (added := summary.get(added_property)):
+                new_total += int(added)
+            if new_total >= 0 and (removed := summary.get(removed_property)):
+                new_total -= int(removed)
+            if new_total >= 0:
+                summary[total_property] = str(new_total)
+
+    _update_totals(
+        total_property='total-data-files',
+        added_property='added-data-files',
+        removed_property='deleted-data-files',
+    )
+    _update_totals(
+        total_property='total-delete-files',
+        added_property='added-delete-files',
+        removed_property='removed-delete-files',
+    )
+    _update_totals(
+        total_property='total-records',
+        added_property='added-records',
+        removed_property='deleted-records',
+    )
+    _update_totals(
+        total_property='total-files-size',
+        added_property='added-files-size',
+        removed_property='deleted-files-size',
```

Comment:
I think this should be [`removed-files-size`](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/SnapshotSummary.java#L44).

Reply:
Yes, thanks!